src/commands/reword.rs

```diff
@@ -0,0 +1,358 @@
+    let result = execute_rebase_plan(effects, git_run_info, &repo, &rebase_plan, &execute_options)?;
+
+    let exit_code = match result {
+        ExecuteRebasePlanResult::Succeeded { rewritten_oids } => {
+            render_status_report(&repo, effects, &commits, &rewritten_oids)?;
+
+            0
+        }
+        ExecuteRebasePlanResult::DeclinedToMerge { merge_conflict: _ } => {
+            writeln!(
+                effects.get_error_stream(),
+                "BUG: Merge conflict detected, but rewording shouldn't cause any conflicts."
+            )?;
+
+            1
+        }
+        ExecuteRebasePlanResult::Failed { exit_code } => exit_code,
+    };
+
+    Ok(exit_code)
+}
+
+/// Turn a list of ref-ish strings into a list of Commits.
+fn resolve_commits_from_hashes<'repo>(
+    repo: &'repo Repo,
+    dag: &mut Dag,
+    effects: &Effects,
+    hashes: Vec<String>,
+) -> eyre::Result<Option<Vec<Commit<'repo>>>> {
+    let hashes = if hashes.is_empty() {
+        vec!["HEAD".to_string()]
+    } else {
+        hashes
+    };
+
+    let commits = resolve_commits(effects, repo, dag, hashes)?;
+    let commits = match commits {
+        ResolveCommitsResult::Ok { commits } => commits,
+        ResolveCommitsResult::CommitNotFound { commit: hash } => {
+            writeln!(effects.get_output_stream(), "Commit not found: {}", hash)?;
+            return Ok(None);
+        }
+    };
+    Ok(Some(commits))
+}
+
+/// The result of building the reword message.
+#[must_use]
+pub enum BuildRewordMessageResult {
+    /// The reworded message was built successfully.
+    Succeeded {
+        /// The reworded messages for each commit.
+        messages: HashMap<NonZeroOid, String>,
+    },
+
+    /// The reworded message matches the original message.
+    IdenticalMessage,
+
+    /// The reworded message was empty.
+    EmptyMessage,
+}
+
+/// Builds the message(s) that will be used for rewording. These are mapped from each commit's
+/// NonZeroOid to the relevant message.
+#[instrument]
+fn build_messages(
+    messages: &[String],
+    commits: &[Commit],
+    comment_char: Option<u8>,
+) -> eyre::Result<BuildRewordMessageResult> {
+    let message = messages.join("\n\n").trim().to_string();
+
+    let (message, load_editor) = if message.is_empty() {
+        let message_for_editor = match commits.to_vec().as_slice() {
+            [commit] => match commit.get_message_raw()?.into_string() {
+                Ok(msg) => msg,
+                _ => eyre::bail!(
+                    "###
+                    Error decoding original message for commit: {:?}.
+                    Stopping; nothing reworded.
+                    ###",
+                    commit.get_oid()
+                ),
+            },
+            _ => {
+                // TODO(bulk edit) build a bulk edit message for multiple commits
+                format!(
+                    "{} Enter a commit message to apply to {} commits",
+                    comment_char.unwrap(),
```

Comment:
Won't the leading comment here be a number like `20` instead of the actual character? Oh, I guess this is the bulk-editing case, so it's not fleshed out yet.

Reply:
🤦 Yes. The dangers of pushing code late at night. I've fixed this, but also rewritten this whole message in my latest round. Thanks for catching this.